### ISSUE TITLE
Recyclerview scroling to top when items changed.

### ISSUE THREAD

**Issue by evant** [evant/binding-collection-adapter#30]
Using a recyclerview with an ObservableList for the data. When items are added, the recyclerview scrolls back to the top. I suspect this is because the whole adapter is being incorrectly replaced. This happens in `1.0.0` but not `0.16`

**Comment by thuytrinh** [comment]
`ViewDataBinding` will register a callback on `ObservableList` that whenever there's any change (e.g. adding, removal) on the list, it will re-invoke the binding method `setAdapter()`.

**Comment by jurenovic** [comment]
Hey, any news on this one. I have a working solution, but i still notices one or two times that this happened again and can't figure it out. See attached bindings code:

```java
// RecyclerView
@SuppressWarnings("unchecked")
@BindingAdapter(value = {"itemView", "items", "adapter", "itemIds"}, requireAll = false)
public static <T> void setAdapter(RecyclerView recyclerView, ItemViewArg<T> arg, List<T> items, BindingRecyclerViewAdapterFactory factory, BindingRecyclerViewAdapter.ItemIds<T> itemIds) {
  if (arg == null) {
    throw new IllegalArgumentException("itemView must not be null");
  }
  if (factory == null) {
    factory = BindingRecyclerViewAdapterFactory.DEFAULT;
  }
  BindingRecyclerViewAdapter<T> adapter = (BindingRecyclerViewAdapter<T>) recyclerView.getAdapter();
  if (adapter == null) {
    adapter = factory.create(recyclerView, arg);
    adapter.setItemIds(itemIds);
    recyclerView.setAdapter(adapter);
  } else if (!adapter.getItemViewArg().equals(arg)) {
    adapter.notifyDataSetChanged();
  }
  adapter.setItems(items);
}
```

Don't know if this is something you want as a solution.

**Comment by spaget** [comment]
Same issue. Any news ?

**Comment by evant** [comment]
I've released `1.0.1` which reverted dynamic itemView support since it was causing the issue. It may come back in the future if I can find a good way to support it.
